src/Conversions/FileResponsiveImageManipulator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Spatie\MediaLibrary\Conversions;
+
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Spatie\MediaLibrary\ResponsiveImages\Jobs\GenerateResponsiveImagesJob;
+
+class FileResponsiveImageManipulator
+{
+    public function createDerivedFiles(Media $media, array $only = [], bool $onlyMissing = false)
+    {
+        $processFiles = true;
+
+        if (
+            (count($only) && !in_array($media->collection_name, $only)) ||
+            ($onlyMissing && count($media->responsive_images))
```

Comment:
Refactor this complex conditional in separate ifs

Here's an example: https://spatie.be/videos/readable-laravel/refactor-complex-conditionals

Reply:
i separated the 2 conditions & added some comments for clarification